assets/js/components/UserMenu.js

```diff
@@ -0,0 +1,183 @@
+ */
+import { Fragment, useState, useRef, useEffect, useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { getSiteKitAdminURL, clearWebStorage } from '../util';
+import Dialog from './dialog';
+import Button from './button';
+import Menu from './menu';
+import Modal from './modal';
+import { STORE_NAME as CORE_SITE } from '../googlesitekit/datastore/site/constants';
+import { STORE_NAME as CORE_USER } from '../googlesitekit/datastore/user/constants';
+
+const { useSelect } = Data;
+
+function UserMenu() {
+	const proxyPermissionsURL = useSelect( ( select ) => select( CORE_SITE ).getProxyPermissionsURL() );
+	const userEmail = useSelect( ( select ) => select( CORE_USER ).getEmail() );
+	const userPicture = useSelect( ( select ) => select( CORE_USER ).getPicture() );
+	const [ dialogActive, toggleDialog ] = useState( false );
+	const [ menuOpen, toggleMenu ] = useState( false );
+	const menuButtonRef = useRef();
+	const menuRef = useRef();
+
+	useEffect( () => {
+		global.addEventListener( 'mouseup', handleMenuClose );
+		global.addEventListener( 'keyup', handleMenuClose );
+		global.addEventListener( 'keyup', handleDialogClose );
+
+		return () => {
+			global.removeEventListener( 'mouseup', handleMenuClose );
+			global.removeEventListener( 'keyup', handleMenuClose );
+			global.removeEventListener( 'keyup', handleDialogClose );
+		};
+	}, [] );
+
+	const handleMenu = useCallback( () => {
+		toggleMenu( ! menuOpen );
+	}, [ menuOpen ] );
+
+	const handleMenuClose = ( e ) => {
+		if (
+			( ( 'keyup' === e.type && 27 === e.keyCode ) || 'mouseup' === e.type ) &&
+			! menuButtonRef.current.buttonRef.current.contains( e.target ) &&
+			! menuRef.current.menuRef.current.contains( e.target )
+		) {
+			toggleMenu( false );
+		}
+	};
+
+	const handleDialog = useCallback( () => {
+		toggleDialog( ! dialogActive );
+		toggleMenu( false );
+	}, [ dialogActive ] );
+
+	const handleMenuItemSelect = useCallback( ( index, e ) => {
+		if (
+			( ( 'keydown' === e.type && (
+				13 === e.keyCode || // Enter
+					32 === e.keyCode // Space
+			) ) ||
+				'click' === e.type // Mouse
+			) ) {
+			switch ( index ) {
+				case 0:
+					handleDialog();
+					break;
+				case 1:
+					if ( ! proxyPermissionsURL ) {
+						return;
+					}
+					global.location.assign( proxyPermissionsURL );
+					break;
+				default:
+					handleMenu();
+			}
+		}
+	}, [ proxyPermissionsURL, handleMenu, handleDialog ] );
+
+	const handleDialogClose = ( e ) => {
+		if ( 27 === e.keyCode ) {
+			toggleDialog( false );
+			toggleMenu( false );
+		}
+	};
+
+	// Log the user out if they confirm the dialog.
+	const handleUnlinkConfirm = useCallback( async () => {
+		// Close the modal.
+		toggleDialog( false );
+
+		// Clear caches.
+		clearWebStorage();
+
+		// Navigate back to the splash screen to reconnect.
+		document.location = getSiteKitAdminURL(
```

Comment:
Let's get this URL from the core/site datastore by calling the `getAdminURL` selector with appropriate arguments.